scene/main/node.cpp

```diff
@@ -2570,11 +2570,9 @@ void Node::print_orphan_nodes() {
 }
 
 void Node::queue_delete() {
-	if (is_inside_tree()) {
-		get_tree()->queue_delete(this);
-	} else {
-		SceneTree::get_singleton()->queue_delete(this);
-	}
+	SceneTree *tree = SceneTree::get_singleton();
```

Comment:
If we are going this route, it looks like `get_tree()` would become pointless. So far it seems the API avoided assumptions that there is only one SceneTree singleton. Originally there was only the `get_tree()` option. Then the `else` was added much later.